File: benchmark/corpus/source/java/Uses.java
// Positive cases. Line numbers matter — the manifest refers to them.
public class Uses {                                               // 2
    void ciphers() throws Exception {                             // 3
        Cipher.getInstance("AES/GCM/NoPadding");                  // 4
        Cipher.getInstance("AES/ECB/PKCS5Padding");               // 5
        Cipher.getInstance("DESede/CBC/PKCS5Padding");            // 6
        Cipher.getInstance("RSA/ECB/OAEPPadding");                // 7
    }                                                             // 8
    void digests() throws Exception {                             // 9
        MessageDigest.getInstance("MD5");                         // 10
        MessageDigest.getInstance("SHA-256");                     // 11
        MessageDigest.getInstance("SHA3-512");                    // 12
    }                                                             // 13
    void signatures() throws Exception {                          // 14
        Signature.getInstance("SHA256withRSA");                   // 15
        Signature.getInstance("SHA256withECDSA");                 // 16
    }                                                             // 17
    void agreement() throws Exception {                           // 18
        KeyAgreement.getInstance("ECDH");                         // 19
    }                                                             // 20
    void generation() throws Exception {                          // 21
        KeyPairGenerator.getInstance("RSA");                      // 22
    }                                                             // 23
    void mac() throws Exception {                                 // 24
        Mac.getInstance("HmacSHA256");                            // 25
    }                                                             // 26
    void random() {                                               // 27
        new java.util.Random();                                   // 28
    }                                                             // 29
}                                                                 // 30

File: benchmark/corpus/source/python/uses.py
"""Positive cases: every line here is a real cryptographic call site.

Line numbers matter — the manifest refers to them. Do not reformat.
"""

import hashlib                                                    # 6
import hmac                                                       # 7

from cryptography.hazmat.primitives import hashes                 # 9
from cryptography.hazmat.primitives.asymmetric import ec, padding, rsa   # 10
from cryptography.hazmat.primitives.ciphers import algorithms, modes     # 11

hashlib.md5(b"legacy checksum")                                   # 13
hashlib.sha1(b"legacy signature")                                 # 14
hashlib.sha256(b"modern")                                         # 15
hashlib.sha512(b"modern")                                         # 16
hashlib.sha3_256(b"sponge")                                       # 17
hashlib.sha3_512(b"sponge")                                       # 18
hashlib.blake2b(b"fast")                                          # 19
hashlib.blake2s(b"fast")                                          # 20
hashlib.shake_256(b"xof")                                         # 21

rsa.generate_private_key(public_exponent=65537, key_size=2048)    # 23
rsa.generate_private_key(public_exponent=65537, key_size=1024)    # 24

padding.PSS(mgf=None, salt_length=32)                             # 26
padding.OAEP(mgf=None, algorithm=None, label=None)                # 27
padding.PKCS1v15()                                                # 28

ec.SECP256R1()                                                    # 30
ec.SECP384R1()                                                    # 31

algorithms.AES(b"0" * 32)                                         # 33
algorithms.TripleDES(b"0" * 24)                                   # 34
algorithms.ChaCha20(b"0" * 32, b"0" * 16)                         # 35
modes.ECB()                                                       # 36
modes.GCM(b"0" * 12)                                              # 37

hashes.SHA3_512()                                                 # 39
hashes.BLAKE2b(64)                                                # 40
